src/malloy/runtime.py

```diff
@@ -125,19 +125,16 @@ async def get_sql(self, named_query=None, query=None):
       else:
         raise ValueError("Channel not in ready state", state)
 
-    return [self._sql, self._connections]
+    return [self._sql, self._connection]
 
   async def run(self, connection, query=None, named_query=None):
-    [sql, connections] = await self.get_sql(query=query,
+    [sql, connection] = await self.get_sql(query=query,
```

Comment:
I just realized I'm just completely overwriting the parameter here, I'm not sure it's necessary anymore? Or I can make it override only like before.

Reply:
Yeah maybe we can remove connection from the run signature, if we're actually getting the information back from the compiler.

If they want to run the sql against a different connection they could call get_sql.  Or we could make it optional if someone wants to compile the query against a different connection than running for some reason.